### PR TITLE
Fix mismatched delete/free in OSQPModel

### PR DESCRIPTION
**Pull request by Jack-ReframeSystems**
Previously we were calling delete on the CSC matrix pointer but its malloc'd, we should be calling "free" instead. We also don't want to free any of the subfield (using something like `csc_spfree`) since we manage the individual fields in OSQPModel vectors.

### CODE REVIEW

**Review by Levi-Armstrong — CHANGES_REQUESTED** [review]
LGTM, Could can you run clang format? There is a script .run-clang-format that you may run.

**Comment by Jack-ReframeSystems** [comment]
> LGTM, Could can you run clang format? There is a script .run-clang-format that you may run.

Do you mind if I change the clang format script to use a newer version of clang? Its using clang 8 at the moment but I can't get that on Ubuntu Jammy, I would like to change it to use the clang default version on the system. 

I'll update the PR with this and you can let me know if you have concerns with this.

File: .run-clang-format [hunk 1]
```diff
@@ -1,2 +1,2 @@
 #!/bin/bash
-find . -type f -regex '.*\.\(cpp\|hpp\|cc\|cxx\|h\|hxx\)' -exec clang-format-8 -style=file -i {} \;
```
Comment (Levi-Armstrong) [inline .run-clang-format:2]: Just for consistency please revert change.

**Comment by Levi-Armstrong** [comment]
replaced by #399

**Comment by rjoomen** [comment]
Good catch!

**Pull request closed without merging**